tests/MiddlewareTest.php

```diff
@@ -54,6 +54,23 @@ public function testNextHandler()
     }
 
 
+    /**
+     * Ensure `$next` will call the next handler.
+     */
+    public function testNextHandlerInvoke()
+    {
+        $this->middleware
+            ->add(function ($next) {
+                $next();
+            })
+            ->add(function ($next) {
+                $this->assertTrue(true);
+            });
+
+        call_user_func($this->middleware);
```

Comment:
Is there any advantage of invoking it with `call_user_func` over just calling it like this: `$this->middleware()`?

Reply:
Trying it that way makes PHP look for a function by that name, instead of a closure bound to the property of that name and therefore fails. The only other way I know it works is `($this->middleware)()` which seems worse than `call_user_func()` to me

Reply:
Ugh, ok, makes sense